File: javascript/724. Find Pivot Index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
  
    if(nums.length == 0){
        return -1;
    }
    let sum = nums.reduce((sum, number) => sum+number);
    let leftSum = 0;
    for(let index = 0; index < nums.length; index++){
        if((sum - nums[index])/2 == leftSum){
            return index;
        }
        leftSum += nums[index];
    }
    return -1;
};

File: java/724. Find Pivot Index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;
        int leftSum = 0;
        
        for(int number:nums){
            sum += number;
        }
        
        for(int index = 0; index < nums.length; index++){
            int diff = sum - nums[index];
            if(diff/2 == leftSum && diff%2 == 0){
                return index;
            }
            leftSum += nums[index];
        }
        return -1;
    }
}
